dotCMS/src/main/java/com/dotcms/rest/api/v1/secret/SiteViewPaginator.java

```diff
@@ -0,0 +1,169 @@
+    }
+
+    /**
+     * This custom getItem implementation will extract join and apply filtering sorting and pagination.
+     * @param user user to filter
+     * @param filter extra filter parameter
+     * @param limit Number of items to return
+     * @param offset offset
+     * @param orderBy This param is ignored.
+     * @param direction This param is ignored.
+     * @param extraParams This param is ignored.
+     * @return SiteView pageItems.
+     * @throws PaginationException
+     */
+    @Override
+    public PaginatedArrayList<SiteView> getItems(final User user, final String filter,
+            final int limit, final int offset,
+            final String orderBy, final OrderDirection direction,
+            final Map<String, Object> extraParams) throws PaginationException {
+        try {
+            //get all sites. Even though this comes from the index. it is permissions driven.
+            final List<String> allSitesIdentifiers = getHostIdentifiers(user, filter);
+
+            final long totalCount = allSitesIdentifiers.size();
+
+            //This values are fed from the outside through the serviceIntegrationAPI.
+            final Set<String> sitesWithConfigurations = configuredSitesSupplier.get();
+            final LinkedHashSet<String> allSites = new LinkedHashSet<>(allSitesIdentifiers);
+
+            //By doing this we remove from the configured-sites collection whatever sites didn't match the search.
+            //If it isn't part of the search results also discard from the configured sites we intent to show.
+            final LinkedHashSet<String> configuredSites = sitesWithConfigurations.stream()
+                    .filter(allSites::contains)
+                    .collect(Collectors.toCollection(LinkedHashSet::new));
+
+            final List<String> finalList = join(configuredSites, allSitesIdentifiers).stream()
+                    .skip(offset).limit(limit).collect(Collectors.toList());
+
+            //And finally load from db and map into the desired view.
+            final List<SiteView> siteViews = finalList.stream().map(id -> {
+                try {
+                    return hostAPI.find(id, user, false);
+                } catch (DotDataException | DotSecurityException e) {
+                    Logger.error(SiteViewPaginator.class, e);
+                }
+                return null;
+            }).filter(Objects::nonNull).map(host -> {
+                final boolean configured = configuredSites.contains(host.getIdentifier());
+                return new SiteView(host.getIdentifier(), host.getName(), configured);
+            }).collect(Collectors.toList());
+
+            //And then we're done and out of here.
+            final PaginatedArrayList<SiteView> paginatedArrayList = new PaginatedArrayList<>();
+            paginatedArrayList.setTotalResults(totalCount);
+            paginatedArrayList.addAll(siteViews);
+            return paginatedArrayList;
+
+        } catch (Exception e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4960082)